app/sequencescape/sequencescape/api/v2/tube_rack.rb

```diff
@@ -1,25 +1,42 @@
 # frozen_string_literal: true
 
-# Tube racks can be barcoded, and contain tubes at defined locations.
+# Tube racks can be barcoded, and contain racked tubes at defined locations.
 class Sequencescape::Api::V2::TubeRack < Sequencescape::Api::V2::Base
   include WellHelpers::Extensions
   include Sequencescape::Api::V2::Shared::HasPurpose
   include Sequencescape::Api::V2::Shared::HasBarcode
+  include Sequencescape::Api::V2::Shared::HasPolyMetadata
 
-  has_many :racked_tubes
+  self.tube_rack = true
 
-  property :created_at, type: :time
-  property :updated_at, type: :time
+  # This is needed in order for the URL helpers to work correctly
+  def to_param
+    uuid
+  end
+
+  #
+  # Override the model used in form/URL helpers
+  # to allow us to treat old and new api the same
+  #
+  # @return [ActiveModel::Name] The resource behaves like a Limber::TubeRack
+  #
+  def model_name
+    ::ActiveModel::Name.new(Limber::TubeRack, false)
+  end
+
+  has_many :racked_tubes, class_name: 'Sequencescape::Api::V2::RackedTube'
+  has_many :parents, class_name: 'Sequencescape::Api::V2::Asset' # Having issues with polymorphism, temporary class
 
   property :name
   property :size
   property :number_or_rows
   property :number_of_columns
+  property :tuberack_barcode
```

Comment:
I would just go for `barcode`... but I guess you're following a convention from tube and plate classes?

Reply:
Doesn't appear to be needed so removed it. Class uses common HasBarcode concern.